File: persistencia/src/main/java/co/edu/uniquindio/proyecto/dto/ProductoGetDTO.java
package co.edu.uniquindio.proyecto.dto;

import co.edu.uniquindio.proyecto.entidades.*;
import lombok.*;

import javax.persistence.*;
import javax.validation.constraints.Future;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.PositiveOrZero;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class ProductoGetDTO {

    private Integer codigo;
    private String usuarioCedula;
    private String usuarioNombre;
    private Integer numeroFavoritos;
    private Boolean isActivo;
    private String imagen;
    private String nombre;
    private String descripcion;
    private Double precio;
    private Integer unidades;
    private Estado_Producto estado;
    private Set<Categoria_Producto> categorias = new HashSet<>();
    private List<ComentarioGetDTO> comentarios= new ArrayList<>();
    private Boolean isDisponible;
    private LocalDateTime fechaLimite;

}

File: my-app/src/app/modelo/producto-get-dto.ts
import { ComentarioPost } from "./comentario-post";

export class ProductoGetDTO {
  codigo: number;
  usuarioCedula: string;
  usuarioNombre: string;
  numeroFavorito: number;
  isActivo: boolean;
  UrlImagen: string;
  nombre: string;
  descripcion: string;
  precio: number;
  unidades: number;
  estado_producto: string[];
  fechaLimite: Date;
  categoria: string[];
  isDisponible: boolean;
  comentarios: ComentarioPost[];

  constructor(
    codigo: number,
    usuarioCedula: string,
    usuarioNombre: string,
    numeroFavorito: number,
    isActivo: boolean,
    UrlImagen: string,
    nombre: string,
    descripcion: string,
    precio: number,
    unidades: number,
    estado_producto: string[],
    fechaLimite: Date,
    categoria: string[],
    isDisponible: boolean,
    comentarios: ComentarioPost[]
  ) {
    this.codigo = codigo;
    this.usuarioCedula = usuarioCedula;
    this.usuarioNombre = usuarioNombre;
    this.numeroFavorito = numeroFavorito;
    this.isActivo = isActivo;
    this.UrlImagen = UrlImagen;
    this.nombre = nombre;
    this.descripcion = descripcion;
    this.precio = precio;
    this.unidades = unidades;
    this.estado_producto = estado_producto;
    this.fechaLimite = fechaLimite;
    this.categoria = categoria;
    this.isDisponible = isDisponible;
    this.comentarios = comentarios;
  }
}
